axelor-human-resource/src/main/java/com/axelor/apps/hr/service/leavereason/LeaveReasonDomainServiceImpl.java

```diff
@@ -0,0 +1,63 @@
+package com.axelor.apps.hr.service.leavereason;
+
+import com.axelor.apps.hr.db.Employee;
+import com.axelor.apps.hr.db.LeaveLine;
+import com.axelor.apps.hr.db.LeaveReason;
+import com.axelor.apps.hr.db.repo.LeaveReasonRepository;
+import com.axelor.auth.AuthUtils;
+import com.axelor.auth.db.User;
+import com.axelor.utils.helpers.StringHelper;
+import com.google.inject.Inject;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class LeaveReasonDomainServiceImpl implements LeaveReasonDomainService {
+
+  protected final LeaveReasonRepository leaveReasonRepository;
+
+  @Inject
+  public LeaveReasonDomainServiceImpl(LeaveReasonRepository leaveReasonRepository) {
+    this.leaveReasonRepository = leaveReasonRepository;
+  }
+
+  @Override
+  public String getLeaveReasonDomain(LeaveReason leaveReason, Employee employee) {
+    return "self.id IN (" + StringHelper.getIdListString(getLeaveReasons(employee)) + ")";
+  }
+
+  protected Set<LeaveReason> getLeaveReasons(Employee employee) {
+    StringBuilder filter = new StringBuilder("self.leaveReasonTypeSelect = 2");
```

Comment:
I don't understand why we have to fetch leave reasons, find the ids, then build a domain instead of directly returning the base JPQL filter as a domain ? It seems like an unnecessary step.

Also since it is in java, we should use static variable for select values.